source/javascripts/pages/PipelinesPage/components/PipelineCanvas/GraphPipelineCanvas/components/WorkflowNode/WorkflowNode.tsx

```diff
@@ -59,10 +64,32 @@ const WorkflowNode = ({ data: { pipelineId }, id, zIndex, selected }: Props) =>
   }, [deleteElements, isGraphPipelinesEnabled]);
 
   const containerProps = useMemo(
-    () => ({ ...defaultStyle, ...(selected ? selectedStyle : {}), ...(isHover ? hoverStyle : {}) }),
-    [selected, isHover],
+    () => ({ ...defaultStyle, ...(selected ? selectedStyle : {}), ...(hovered ? hoveredStyle : {}) }),
+    [selected, hovered],
   );
 
+  const highlighted = selected || hovered;
+
+  useEffect(() => {
+    setEdges((edges) => {
+      return edges.map((edge) => {
+        if (edge.target === id) {
+          const newEdge = { ...edge, zIndex: DEFAULT_GRAPH_EDGE_ZINDEX, data: { ...edge.data, highlighted } };
+
+          if (edge.selected) {
+            newEdge.zIndex = SELECTED_GRAPH_EDGE_ZINDEX;
+          } else if (highlighted) {
+            newEdge.zIndex = HIGHLIGHTED_GRAPH_EDGE_ZINDEX;
+          }
```

Comment:
Doesn't the edge itself handle these already?

Reply:
Nope. Unfortunately...